### ISSUE TITLE
Hide task definitions that are part of the CTR application from task list page

### ISSUE THREAD

**Issue by ilayaperumalg** [spring-cloud/spring-cloud-dataflow-ui#1306]
The individual task definitions that are part of the CTR can be hidden from the task list page from the UI.  This would allow the accidental launch of those applications outside the scope of the CTR.

**Comment by oodamien** [comment]
@ilayaperumalg Instead of doing that on the UI side (which will break the pagination), can we add a parameter in the service (`GET /tasks/definitions`) to exclude this tasks?

**Comment by jvalkeal** [comment]
Yeah, we should not just blindly hide as those are real definitions.

**Comment by jvalkeal** [comment]
And besides, from a task definition point of view, we don't have info about parent definition so relying on a prefix just won't work. i.e. `task`, `task-t1` and `task-t2`. If you then create a separate task `task-xxx` there's no way to know unless you start to do a complex analysis of all existing task defs. Don't think we wanna go down that route.

**Comment by sabbyanandan** [comment]
This requires adjustments to the server-side API.

**Comment by sabbyanandan** [comment]
Doing this in the GUI will be expensive. We need the API to return the root-level Task definition, and children attached to it. We can then TREE it on the GUI eventually.
